core/controllers/mappings.py

```diff
@@ -0,0 +1,274 @@
+from dataclasses import dataclass, field
```

Comment:
It might be good to put a module docstring in here? Just so it's immediately obvious what the module is for?

Reply:
good shout